src/Language/PureScript/CoreFn/Desugar.hs

```diff
@@ -38,11 +38,27 @@ moduleToCoreFn env (A.Module modSS coms mn decls (Just exps)) =
   let imports = mapMaybe importToCoreFn decls ++ fmap (ssAnn modSS,) (findQualModules decls)
       imports' = dedupeImports imports
       exps' = ordNub $ concatMap exportToCoreFn exps
+      reExps = M.map ordNub $ M.unionsWith (++) (map (reExportsToCoreFn mn) exps)
```

Comment:
Should we be passing all of `exps` to this function? I understand that you have the `mn == mn'` guard in `reExportsToCoreFn` preventing normal exports being regarded as re-exports, but I feel like it might be simpler to only pass re-exports in in the first place. Would it be possible to write this via something like
`mapMaybe (fmap reExportsToCoreFn . toReExportRef)`, where `toReExportRef` is defined as something like
```
toReExportRef ref = case ref of
  A.ReExportRef _ src ref -> Just (src, ref)
  _ -> Nothing
```

Reply:
The reason I'm passing all of `exps` to `reExportsToCoreFn` is because I call the function recursively to handle the descending into `ref`. I can reformulate it in terms of `exportToCoreFn` though and only pass in the re-exports.

Reply:
Right - I would prefer to avoid recursion if we can. Did you come across cases in which there is a ReExportRef of a ReExportRef? I don't think that should happen but I might be wrong.

Reply:
Could that happen when re-exporting a re-export?

Reply:
I guess I can just write a test and find out 😄

Reply:
The types don't rule it out, but I think what should happen is that re-exports should always be of the form `ReExportRef _ _ ref`, where `ref`'s constructor is something other than `ReExportRef`. See elaborateExports in Sugar.Names, that's where this data is assembled.

Reply:
Multiple layers of re-exports are tracked via the `ExportSource` field.

Reply:
> See elaborateExports in Sugar.Names, that's where this data is assembled

Doesn't look like [`elaborateExports`](https://github.com/purescript/purescript/blob/f6e9a7bb10a9d938bc4c3668f2793bfc54b636e1/src/Language/PureScript/Sugar/Names.hs#L140-L143) will nest `ReExportRef`s. Looks like we're good.